homeassistant/components/lg_netcast/media_player.py

```diff
@@ -211,8 +254,10 @@ def turn_off(self) -> None:
 
     def turn_on(self) -> None:
         """Turn on the media player."""
-        if self._on_action_script:
-            self._on_action_script.run(context=self._context)
+        asyncio.run_coroutine_threadsafe(
+            self._turn_on.async_run(self.hass, self._context),
+            self.hass.loop,
+        ).result()
```

Comment:
If the turn on action is now a coroutine, this can be simplified to:
```
    async def async_turn_on(self) -> None:
        """Turn on the media player."""
        await self._turn_on.async_run(self.hass, self._context)
```